source/tutorials/kubernetes.rst

```diff
@@ -3,354 +3,716 @@
+
+   .. code-block:: bash
 
-Before you install any new packages, update |CL| with the following command:
+      sudo swupd update
 
-.. code-block:: bash
+   Learn about the benefits of having an up-to-date system for cloud
+   orchestration on the :ref:`swupd-guide` page.
 
-   sudo swupd update
+#. Kubernetes, a set of supported :abbr:`CRI (Container Runtime Interface)`
+   runtimes, :abbr:`CNI (Container Network Interface)` and `cloud-native-setup
+   scripts`_ are included in the `cloud-native-basic`_ bundle. Install the
+   cloud-native-basic bundle to get these components:
 
-Install Kubernetes and CRI runtimes
-***********************************
+   .. code-block:: bash
 
-Kubernetes, a set of supported :abbr:`CRI (Container Runtime Interface)`
-runtimes, and networking plugins, are included in the `cloud-native-basic`_
-bundle.
+      sudo swupd bundle-add cloud-native-basic
 
-To install this framework, enter the following command:
 
-.. code-block:: bash
+Set up Kubernetes automatically
+*******************************
 
-   sudo swupd bundle-add cloud-native-basic
+|CL| provides `cloud-native-setup scripts`_ to automate system setup and
+Kubernetes cluster initialization which allows you to get a cluster up and
+running quickly.
 
 .. note::
 
-   For more on networking plugins, see `Install pod network add-on`_.
+   By default, the scripts will update |CL| to the latest version, set up the
+   system as a Kubernetes master-node with **canal for container networking**
+   and **crio for container runtime**, and taint the master node to allow
+   workloads to run on it. Kata is installed as an optional alternative
+   runtime. The script can be configured to use other CNI's and CRI's by
+   following the directions on the `README
+   <https://github.com/clearlinux/cloud-native-setup/blob/master/clr-k8s-examples/README.md>`_.
+   
+   See `What is a Container Network Interface (CNI)?`_ and `What is a
+   Container Runtime Interface (CRI)?`_ for more information.
+
+.. important::
+
+   If network proxy settings are required for Internet connectivity, configure
+   them now because the scripts will propagate proxy configuration based on
+   the running configuration. It is especially important to set the
+   :command:`no_proxy` variable for Kubernetes. See the `Setting proxy servers
+   for Kubernetes`_ section for details.
+
+#. Run the :file:`system-setup.sh` script to configure the |CL| system
+   settings.
+
+   .. code-block:: bash
+
+      sudo /usr/share/clr-k8s-examples/setup_system.sh
+
+#. Stop docker and containerd to avoid conflicting CRIs being detected. The
+   scripts use CRIO for the CRI.
+
+   .. code-block:: bash
+
+      sudo systemctl stop docker
+      sudo systemctl stop containerd
+
+#. Run the :file:`create_stack.sh` script to initialize the Kubernetes node
+   and setup a container network plugin.
+
+   .. code-block:: bash
+
+      sudo /usr/share/clr-k8s-examples/create_stack.sh minimal
```

Comment:
There has to be a better way than this.   Try adding `sudo chown -R $USER $HOME/.kube` as a step to see if it eliminates the need for sudo on kubectl. I gleamed this from https://stackoverflow.com/questions/51280172/how-to-use-kubectl-command-instead-of-sudo-kubectl#51280522

Reply:
Alternatively, This is what the stack suggests to do after the stack is initialized. It might be enough to just `sudo rm -rf $HOME/.kube` and then run the following.

```

  mkdir -p $HOME/.kube
  sudo cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
  sudo chown $(id -u):$(id -g) $HOME/.kube/config

```

Reply:
I'm going to leave this for now because I have observed some inconsistency without `sudo` in the past. Launching with `sudo` should not impact anything since subsequent `sudo` command run. It also makes the whole section consistent with `sudo` prefixed before each command.